### PR TITLE
Try mounting debugfs if not there + a bonus patch

### LINKED ISSUE #278: Help should not require privileges
Currently asking for help as a regular user results in:
```
retis collect --help
Error: Retis does not have CAP_SYS_ADMIN: can't replace BPF programs.
```

ideally, this should not happen, and help should be shown.

### LINKED ISSUE #402: Consider automatic mounting of debugfs if not present
It is painful to manage debugfs mounts in network namespaces, because the mount only exists for the duration of the `ip netns exec`, i.e. if you run `ip netns exec <my-ns> mount -t debugfs debugfs /sys/kernel/debug`, it will not be mounted while running `ip netns exec <my-ns> retis collect` right after.  This can be worked around by executing the shell and both commands inside, but If retis could mount it for us, it would be much easier.

Can be an opt-in behavior.

### PR DESCRIPTION
**Pull request by atenart**
If `debugfs` is not mounted (eg. in containers, namespaces) try mounting it if `--allow-system-changes` is set. The bonus patch is making `retis collect --help` not requiring root privileges.

Closes #402, closes #278.

### CODE REVIEW

**Review by vlrpl — COMMENTED** [review]
the PR lgtm. I only have one question, and other than that it can be merged.

File: retis/src/collect/collector.rs [hunk 1]
```diff
@@ -192,6 +195,26 @@ impl Collectors {
...
+                std::path::Path::new("/sys/kernel/debug"),
+                Some("debugfs"),
+                MsFlags::empty(),
+                None::<&str>,
+            );
+
+            match err {
+                Ok(_) => self.mounted_debugfs = true,
+                Err(errno) => match errno {
+                    Errno::EBUSY => (), // Already mounted.
```
[the remark this answers is not in the document]
Comment (atenart) [reply retis/src/collect/collector.rs:212]: The thing is this is the expected behavior, if we were to add something in the logs (even for debug logs) we would need to check if debugfs is already mounted first and not rely on the above.

[the remark this answers is not in the document]
Comment (vlrpl) [reply retis/src/collect/collector.rs:212]: if the errno semantic is really "already mounted" in this case, what's the difference between testing separately and failing while trying?
The purpose of the log is just to easily allow users to know the tool attempted to mount and then know what to expect on termination (removal).

Speaking about expectations it comes to mind that the help should include that the debugfs gets unmounted upon removal if mounted by the tool.

[the remark this answers is not in the document]
Comment (atenart) [reply retis/src/collect/collector.rs:212]: I see, makes sense. Let's add a message when debugfs is mounted too then.

I'll also fix the help.

**Review by vlrpl — APPROVED** [review]
lgtm, thanks!

**Pull request merged**